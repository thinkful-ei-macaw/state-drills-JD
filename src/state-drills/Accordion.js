import React from 'react';

class Accordion extends React.Component {
  constructor(props){
    super(props)
  }
  
  state = {
    clicked: null
  }

  handleClick(num) {
    console.log('clicked', num);
    this.setState(() => {
      return {clicked: num}
    });

    };
  
  // renderButtons() {
  //   return this.props.sections.map(() => (
  //   <li>
  //     <button>{this.props.sections.title}</button>
  //   </li>
  //   ))
  // }

  //   render() {
  //     return (
  //       <div>
  //         <ul>
  //           {this.renderButtons()}
  //         </ul>
  //       </div>
  //     )
  //   }

  render() {
    if (this.state.clicked === 0) {
      return (
        <ul>
          <li className="section1">
            <button onClick={e => this.handleClick(0)}>{this.props.sections[0].title}</button>
            <p>{this.props.sections[0].content}</p>
          </li>
          <li className="section2">
           <button onClick={e => this.handleClick(1)}>{this.props.sections[1].title}</button>
          </li>
          <li className="section3">
           <button onClick={e => this.handleClick(2)}>{this.props.sections[2].title}</button>
          </li>
      </ul>
      )
    } else if (this.state.clicked === 1) {
    return (
      <ul>
      <li className="section1">
        <button onClick={e => this.handleClick(0)}>{this.props.sections[0].title}</button>
      </li>
      <li className="section2">
       <button onClick={e => this.handleClick(1)}>{this.props.sections[1].title}</button>
       <p>{this.props.sections[1].content}</p>
      </li>
      <li className="section3">
       <button onClick={e => this.handleClick(2)}>{this.props.sections[2].title}</button>
      </li>
  </ul>
  )
  } else if (this.state.clicked === 2) {
    return (
      <ul>
      <li className="section1">
        <button onClick={e => this.handleClick(0)}>{this.props.sections[0].title}</button>
      </li>
      <li className="section2">
       <button onClick={e => this.handleClick(1)}>{this.props.sections[1].title}</button>
      </li>
      <li className="section3">
       <button onClick={e => this.handleClick(2)}>{this.props.sections[2].title}</button>
       <p>{this.props.sections[2].content}</p>
      </li>
  </ul>
    )
  } else {
    return (
      <ul>
      <li className="section1">
        <button onClick={e => this.handleClick(0)}>{this.props.sections[0].title}</button>
      </li>
      <li className="section2">
       <button onClick={e => this.handleClick(1)}>{this.props.sections[1].title}</button>
      </li>
      <li className="section3">
       <button onClick={e => this.handleClick(2)}>{this.props.sections[2].title}</button>
      </li>
  </ul>
    )
  }
  }
}

export default Accordion;