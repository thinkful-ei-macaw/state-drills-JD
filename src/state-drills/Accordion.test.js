// You should write at least 4 snapshot tests for the Accordion component:
// The component renders an empty li when the sections prop is not supplied.
// The component renders no sections as active by default.
// The component opens a clicked section.
// The component only opens the last section when multiple sections have been clicked.

import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe(`Accordion Component`, () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})