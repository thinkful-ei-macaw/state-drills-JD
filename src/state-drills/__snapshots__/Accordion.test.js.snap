// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty given no sections 1`] = `<div />`;
